src/binary_cache.cpp

```diff
@@ -59,12 +60,17 @@ static boost::filesystem::path ComputeUserCachePath()
 {
 #ifdef MIOPEN_CACHE_DIR
     std::string cache_dir = MIOPEN_CACHE_DIR;
+    std::string version;
 
-    std::string version =
-        std::to_string(MIOPEN_VERSION_MAJOR) + "." + std::to_string(MIOPEN_VERSION_MINOR) + "." +
-        std::to_string(MIOPEN_VERSION_PATCH) + "." + MIOPEN_STRINGIZE(MIOPEN_VERSION_TWEAK);
-
+    version = std::to_string(MIOPEN_VERSION_MAJOR) + "." + std::to_string(MIOPEN_VERSION_MINOR) +
+              "." + std::to_string(MIOPEN_VERSION_PATCH) + "." +
+              MIOPEN_STRINGIZE(MIOPEN_VERSION_TWEAK);
     auto p = boost::filesystem::path{miopen::ExpandUser(cache_dir)} / version;
+
+    const char* const custom = miopen::GetStringEnv(MIOPEN_CUSTOM_CACHE_DIR{});
+    if(custom != nullptr && strlen(custom) > 0)
+        p = boost::filesystem::path{miopen::ExpandUser(custom)};
```

Comment:
How so? It is not initialized to the same path. Here we are merely overwriting that initialization with some other path if env var is used, otherwise we leave it as is.

Reply:
When `MIOPEN_CUSTOM_CACHE_DIR` is set, line 68 is useless. It's easy to fix:
```C++
const char* const custom = miopen::GetStringEnv(MIOPEN_CUSTOM_CACHE_DIR{});
const auto p  = (custom != nullptr && strlen(custom) > 0)
    ? boost::filesystem::path{miopen::ExpandUser(custom)}
    : boost::filesystem::path{miopen::ExpandUser(cache_dir)} / version;
```